packages/wdio-cli/src/interface.js

```diff
@@ -0,0 +1,155 @@
+import chalk from 'chalk'
+import cliSpinners from 'cli-spinners'
+import EventEmitter from 'events'
+import CLIInterface from 'wdio-interface'
+
+const clock = cliSpinners['clock']
+
+export default class WDIOCLIInterface extends EventEmitter {
+    constructor (config, specs) {
+        super()
+        this.hasAnsiSupport = chalk.supportsColor.hasBasic
+        this.clockTimer = 0
+        this.specs = specs
+        this.config = config
+        this.jobs = new Map()
+        this.start = Date.now()
+        this.result = {
+            finished: 0,
+            passed: 0,
+            failed: 0
+        }
+        this.messages = {
+            reporter: {}
+        }
+
+        this.interface = new CLIInterface()
+        this.on('job:start', ::this.addJob)
+        this.on('job:end', ::this.clearJob)
+    }
+
+    /**
+     * add job to interface
+     */
+    addJob({ cid, caps, specs }) {
+        this.jobs.set(cid, { caps, specs })
+        this.updateView()
+    }
+
+    /**
+     * clear job from interface
+     */
+    clearJob ({ cid, passed }) {
+        this.jobs.delete(cid)
+        this.result.finished++
+
+        if (passed) {
+            this.result.passed++
+        } else {
+            this.result.failed++
+        }
+
+        this.updateView(true)
+    }
+
+    /**
+     * event handler that is triggered when runner sends up events
+     */
+    onMessage (params) {
+        if (!this.messages[params.origin][params.name]) {
+            this.messages[params.origin][params.name] = []
+        }
+        this.messages[params.origin][params.name].push(params.content)
+    }
+
+    updateView (wasJobCleared) {
+        const isFinished = this.jobs.size === 0
+        const pendingJobs = this.specs.length - this.jobs.size - this.result.finished
+
+        /**
+         * check if environment supports ansi and print a limited update if not
+         */
+        if (!this.hasAnsiSupport && !isFinished) {
+            /**
+             * only update if a job finishes
+             */
+            if (!wasJobCleared) {
+                return
+            }
+
+            const time = clock.frames[this.clockTimer = ++this.clockTimer % clock.frames.length]
```

Comment:
Perhaps rename to timeSpinner or clockSpinner like in the other comment I posted